src/test/kotlin/com/atlassian/performance/tools/ssh/api/SshConnectionTest.kt

```diff
@@ -14,4 +18,18 @@ class SshConnectionTest {
             Assert.assertEquals(sshResult.output, "test\n")
         }
     }
+
+    @Test
+    fun shouldRespectTimeouts() {
+        SshContainer().useConnection { ssh: SshConnection ->
+            val start = Instant.now()
+            try {
+                ssh.execute("sleep 100", Duration.ofSeconds(1), Level.OFF, Level.OFF)
+            }catch (e : Exception){
+                //ignore
+            }
+            val executeDuration = Duration.between( start, Instant.now())
+            Assert.assertTrue(executeDuration < Duration.ofSeconds(10))
```

Comment:
😲

Reply:
It seems that sshj sends SSH_MSG_CHANNEL_CLOSE as per https://www.ietf.org/rfc/rfc4254.txt but it doesn't receive the same message in response. Sshj waits 30 seconds before timeout. I need to Wireshark it to get more details.

Reply:
And it seems to be related to our "Could not get lock /var/lib/dpkg/lock - open (11: Resource temporarily unavailable)" problem. Ubuntu doesn't close the connection and doesn't kill the apt process.

Reply:
Oooh, that's super interesting
I'm glad you found it